tests/unit/models/governing-body-test.js

```diff
@@ -0,0 +1,828 @@
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.false(isValid);
+      assert.ok(model.error, 'model.error should not be undefined');
+      assert.strictEqual(Object.keys(model.error).length, 2);
+      assert.propContains(model.error, {
+        startDate: {
+          message: 'Geen overlap',
+        },
+        endDate: {
+          message: 'Geen overlap',
+        },
+      });
+
+      otherBodiesStub.restore();
+    });
+
+    test('it returns no error when the start and end dates of two governing bodies are identical', async function (assert) {
+      let start = new Date('2024-01-01');
+      let end = new Date('2025-01-01');
+
+      const otherModel = this.store().createRecord('governing-body', {
+        startDate: start,
+        endDate: end,
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: start,
+        endDate: end,
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should be undefined');
+
+      otherBodiesStub.restore();
+    });
+
+    test("it returns no error when the period of another governing body is strictly after the model's period", async function (assert) {
+      const otherModel = this.store().createRecord('governing-body', {
+        startDate: new Date('2024-01-01'),
+        endDate: new Date('2026-01-01'),
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2020-01-01'),
+        endDate: new Date('2023-01-01'),
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should be undefined');
+
+      otherBodiesStub.restore();
+    });
+
+    test("it returns no error when the period of another governing body is strictly before the model's period", async function (assert) {
+      const otherModel = this.store().createRecord('governing-body', {
+        startDate: new Date('2020-01-01'),
+        endDate: new Date('2023-01-01'),
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2024-01-01'),
+        endDate: new Date('2026-01-01'),
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should be undefined');
+
+      otherBodiesStub.restore();
+    });
```

Comment:
Really similar to the previous one

Reply:
I agree that the tests are very similar, but due to the asymmetric definition of overlap that is used I prefer to have tests to explicitly cover all cases.